src/components/todo.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { PropsTodo, Actions, Todo } from '../types';
+import { TodosContext } from '../Store';
+
+type Event =
+  | React.FormEvent<HTMLFormElement>
+  | React.FocusEvent<HTMLInputElement, Element>;
+
+export const TodoItem: React.FC<PropsTodo> = ({ id, title, status }) => {
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState(title);
+  const { todos, setTodos } = useContext(TodosContext);
+
+  function handleClick(action: string) {
+    if (action === Actions.delete) {
+      const newTodos = todos.filter((item: Todo) => item.id !== id);
+
+      setTodos(newTodos);
+    }
+
+    if (action === Actions.updateToDone) {
+      const newTodos = todos.map((item: Todo) => {
+        return item.id === id ? { ...item, completed: true } : item;
+      });
+
+      setTodos(newTodos);
+    }
+
+    if (action === Actions.updateToNotDone) {
+      const newTodos = todos.map((item: Todo) => {
+        return item.id === id ? { ...item, completed: false } : item;
+      });
+
+      setTodos(newTodos);
+    }
+  }
+
+  function handleSubmit(event: Event) {
+    event.preventDefault();
+
+    if (query) {
+      const newList = todos.map((todo: Todo) => {
+        return todo.id === id ? { ...todo, title: query.trim() } : todo;
+      });
+
+      setTodos(newList);
+      setIsFocused(false);
+      setQuery(query);
+    } else {
+      const newList = todos.filter((todo: Todo) => {
+        return todo.id !== id;
+      });
+
+      setTodos(newList);
+      setIsFocused(false);
+      setQuery(title);
+    }
+  }
+
+  function preventSubmit(event: React.KeyboardEvent<HTMLInputElement>) {
+    if (event.key === 'Escape') {
+      setIsFocused(false);
+      setQuery(title);
+    }
+  }
+
+  return (
+    <div data-cy="Todo" className={classNames('todo', { completed: status })}>
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          defaultChecked={status === true ? true : false}
+          onClick={() => {
+            return !status
+              ? handleClick(Actions.updateToDone)
+              : handleClick(Actions.updateToNotDone);
+          }}
+        />
+      </label>
+
+      {!isFocused && (
+        <React.Fragment>
+          <span
+            data-cy="TodoTitle"
+            className="todo__title"
+            onDoubleClick={() => setIsFocused(true)}
+          >
+            {title}
+          </span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => handleClick(Actions.delete)}
+          >
+            ×
+          </button>
+        </React.Fragment>
+      )}
+
+      {isFocused && (
+        <React.Fragment>
```

Comment:
this fragment is redundant here 

https://react.dev/reference/react/Fragment#usage